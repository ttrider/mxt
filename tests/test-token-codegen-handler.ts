import { codegenSetup } from "./utils";
import path from "path";
import fs from "fs";

describe("token", () => {

  const logs: string[] = [];

  afterAll(() => {

    if (logs.length > 0) {

      const filename = path.resolve(__dirname, "__snapshots__", path.basename(__filename) + ".log");
      fs.writeFileSync(filename, logs.join("\n=======================\n\n"));

    }
  });

  test("single element, attribute, token", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}">Hello MXT!</div>
  </template>`;


    const results = codegenSetup(input);

    expect(results).toMatchSnapshot();

  });

  test("single element, multiple attributes,tokens", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}" title="\${title}">Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);
    expect(results).toMatchSnapshot();

  });

  test("single element, multiple tokens per attribute", () => {

    const input =
      `<template id="t01" type="mxt">
    <div id="old" style="color: \${color}; margin: \${margin}" >Hello MXT!</div>
  </template>`;

    const results = codegenSetup(input);

    expect(results).toMatchSnapshot();
  });

  test("multiple elements", () => {
    const input =
      `<template id="t01" type="mxt">
    <div style="color: \${color};">Hello MXT!</div>
    <div title="\${title}">Hello MXT too!</div>
  </template>`;

    const results = codegenSetup(input);

    logs.push(results);
    

    expect(results).toMatchSnapshot();
  })


});
