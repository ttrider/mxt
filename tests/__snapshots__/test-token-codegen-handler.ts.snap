// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`token multiple elements 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \\"\\\\n    <div id=\\\\\\"tagid_4\\\\\\">Hello MXT!</div>\\\\n    <div id=\\\\\\"tagid_5\\\\\\">Hello MXT too!</div>\\\\n  \\";
export function t01(data: any, host?: null | undefined | Element) {
    let disposed = false;
    const { $$mxt$$elements$$, tagid_4$$element, tagid_5$$element } = $$mxt$$initialize$$(t01$$template, [\\"tagid_4\\", \\"tagid_5\\"]);
    tagid_4$$element.id = \\"\\";
    const tagid_4$$autorun = autorun(() => {
        const { color } = data;
        tagid_4$$element.setAttribute(\\"style\\", \`color: \${color};\`);
    });
    tagid_5$$element.id = \\"\\";
    const tagid_5$$autorun = autorun(() => {
        const { title } = data;
        tagid_5$$element.setAttribute(\\"title\\", \`\${title}\`);
    });
    if (host)
        $$mxt$$appendTo$$($$mxt$$elements$$, host);
    return {
        get disposed() {
            return disposed;
        },
        get elements() {
            return $$mxt$$elements$$;
        },
        appendTo: (host: Element) => $$mxt$$appendTo$$($$mxt$$elements$$, host),
        remove: () => $$mxt$$remove$$($$mxt$$elements$$),
        dispose: () => {
            disposed = true;
            $$mxt$$remove$$($$mxt$$elements$$);
            $$mxt$$elements$$.splice(0, $$mxt$$elements$$.length);
            tagid_4$$autorun();
            tagid_5$$autorun();
        }
    };
}
function $$mxt$$initialize$$(template: HTMLTemplateElement, elementIds: string[]) {
    const elements: Element[] = [];
    let child = template.content.firstElementChild;
    while (child) {
        elements.push(child);
        child = child.nextElementSibling;
    }
    const context: any = { $$mxt$$elements$$: elements };
    for (const elementId of elementIds) {
        const element = template.content.getElementById(elementId);
        if (element) {
            context[elementId + \\"$$element\\"] = element;
        }
    }
    return context;
}
function $$mxt$$appendTo$$(elements: Element[], host: Element) {
    for (const el of elements) {
        host.appendChild(el);
    }
}
function $$mxt$$remove$$(elements: Element[]) {
    for (const el of elements) {
        el.remove();
    }
}
"
`;

exports[`token single element, attribute, token 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \\"\\\\n    <div id=\\\\\\"tagid_1\\\\\\">Hello MXT!</div>\\\\n  \\";
export function t01(data: any, host?: null | undefined | Element) {
    let disposed = false;
    const { $$mxt$$elements$$, tagid_1$$element } = $$mxt$$initialize$$(t01$$template, [\\"tagid_1\\"]);
    tagid_1$$element.id = \\"old\\";
    const tagid_1$$autorun = autorun(() => {
        const { color } = data;
        tagid_1$$element.setAttribute(\\"style\\", \`color: \${color}\`);
    });
    if (host)
        $$mxt$$appendTo$$($$mxt$$elements$$, host);
    return {
        get disposed() {
            return disposed;
        },
        get elements() {
            return $$mxt$$elements$$;
        },
        appendTo: (host: Element) => $$mxt$$appendTo$$($$mxt$$elements$$, host),
        remove: () => $$mxt$$remove$$($$mxt$$elements$$),
        dispose: () => {
            disposed = true;
            $$mxt$$remove$$($$mxt$$elements$$);
            $$mxt$$elements$$.splice(0, $$mxt$$elements$$.length);
            tagid_1$$autorun();
        }
    };
}
function $$mxt$$initialize$$(template: HTMLTemplateElement, elementIds: string[]) {
    const elements: Element[] = [];
    let child = template.content.firstElementChild;
    while (child) {
        elements.push(child);
        child = child.nextElementSibling;
    }
    const context: any = { $$mxt$$elements$$: elements };
    for (const elementId of elementIds) {
        const element = template.content.getElementById(elementId);
        if (element) {
            context[elementId + \\"$$element\\"] = element;
        }
    }
    return context;
}
function $$mxt$$appendTo$$(elements: Element[], host: Element) {
    for (const el of elements) {
        host.appendChild(el);
    }
}
function $$mxt$$remove$$(elements: Element[]) {
    for (const el of elements) {
        el.remove();
    }
}
"
`;

exports[`token single element, multiple attributes,tokens 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \\"\\\\n    <div id=\\\\\\"tagid_2\\\\\\">Hello MXT!</div>\\\\n  \\";
export function t01(data: any, host?: null | undefined | Element) {
    let disposed = false;
    const { $$mxt$$elements$$, tagid_2$$element } = $$mxt$$initialize$$(t01$$template, [\\"tagid_2\\"]);
    tagid_2$$element.id = \\"old\\";
    const tagid_2$$autorun = autorun(() => {
        const { color, title } = data;
        tagid_2$$element.setAttribute(\\"style\\", \`color: \${color}\`);
        tagid_2$$element.setAttribute(\\"title\\", \`\${title}\`);
    });
    if (host)
        $$mxt$$appendTo$$($$mxt$$elements$$, host);
    return {
        get disposed() {
            return disposed;
        },
        get elements() {
            return $$mxt$$elements$$;
        },
        appendTo: (host: Element) => $$mxt$$appendTo$$($$mxt$$elements$$, host),
        remove: () => $$mxt$$remove$$($$mxt$$elements$$),
        dispose: () => {
            disposed = true;
            $$mxt$$remove$$($$mxt$$elements$$);
            $$mxt$$elements$$.splice(0, $$mxt$$elements$$.length);
            tagid_2$$autorun();
        }
    };
}
function $$mxt$$initialize$$(template: HTMLTemplateElement, elementIds: string[]) {
    const elements: Element[] = [];
    let child = template.content.firstElementChild;
    while (child) {
        elements.push(child);
        child = child.nextElementSibling;
    }
    const context: any = { $$mxt$$elements$$: elements };
    for (const elementId of elementIds) {
        const element = template.content.getElementById(elementId);
        if (element) {
            context[elementId + \\"$$element\\"] = element;
        }
    }
    return context;
}
function $$mxt$$appendTo$$(elements: Element[], host: Element) {
    for (const el of elements) {
        host.appendChild(el);
    }
}
function $$mxt$$remove$$(elements: Element[]) {
    for (const el of elements) {
        el.remove();
    }
}
"
`;

exports[`token single element, multiple tokens per attribute 1`] = `
"import { autorun } from \\"mobx\\";
const t01$$template = document.createElement(\\"template\\");
t01$$template.innerHTML = \\"\\\\n    <div id=\\\\\\"tagid_3\\\\\\">Hello MXT!</div>\\\\n  \\";
export function t01(data: any, host?: null | undefined | Element) {
    let disposed = false;
    const { $$mxt$$elements$$, tagid_3$$element } = $$mxt$$initialize$$(t01$$template, [\\"tagid_3\\"]);
    tagid_3$$element.id = \\"old\\";
    const tagid_3$$autorun = autorun(() => {
        const { color, margin } = data;
        tagid_3$$element.setAttribute(\\"style\\", \`color: \${color}; margin: \${margin}\`);
    });
    if (host)
        $$mxt$$appendTo$$($$mxt$$elements$$, host);
    return {
        get disposed() {
            return disposed;
        },
        get elements() {
            return $$mxt$$elements$$;
        },
        appendTo: (host: Element) => $$mxt$$appendTo$$($$mxt$$elements$$, host),
        remove: () => $$mxt$$remove$$($$mxt$$elements$$),
        dispose: () => {
            disposed = true;
            $$mxt$$remove$$($$mxt$$elements$$);
            $$mxt$$elements$$.splice(0, $$mxt$$elements$$.length);
            tagid_3$$autorun();
        }
    };
}
function $$mxt$$initialize$$(template: HTMLTemplateElement, elementIds: string[]) {
    const elements: Element[] = [];
    let child = template.content.firstElementChild;
    while (child) {
        elements.push(child);
        child = child.nextElementSibling;
    }
    const context: any = { $$mxt$$elements$$: elements };
    for (const elementId of elementIds) {
        const element = template.content.getElementById(elementId);
        if (element) {
            context[elementId + \\"$$element\\"] = element;
        }
    }
    return context;
}
function $$mxt$$appendTo$$(elements: Element[], host: Element) {
    for (const el of elements) {
        host.appendChild(el);
    }
}
function $$mxt$$remove$$(elements: Element[]) {
    for (const el of elements) {
        el.remove();
    }
}
"
`;
